plugin/healthchecker/setup.go

```diff
@@ -0,0 +1,80 @@
+package healthchecker
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/coredns/caddy"
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+)
+
+func init() {
+	plugin.Register(pluginName, setup)
+}
+
+func setup(c *caddy.Controller) error {
+	c.Next()
+	filter, err := filterParamsParse(c)
+	if err != nil {
+		return err
+	}
+
+	dnsserver.GetConfig(c).AddPlugin(func(next plugin.Handler) plugin.Handler {
+		return HealthChecker{
+			Next:   next,
+			filter: filter,
+		}
+	})
+
+	return nil
+}
+
+func filterParamsParse(c *caddy.Controller) (*HealthCheckFilter, error) {
+	var checker Checker
+	var err error
+	args := c.RemainingArgs()
+	if len(args) < 4 {
+		return nil, plugin.Error(pluginName,
+			fmt.Errorf("the following format is supported: HEALTHCHECK_METHOD CACHE_SIZE "+
+				"HEALTHCHECK_INTERVAL_IN_MS ORIGIN. [NAME_FILTER1 NAME_FILTER2]"))
+	}
+
+	if strings.Contains(args[0], "http") {
+		checker, err = NewHttpChecker(args[0])
+		if err != nil {
+			return nil, err
+		}
+	}
+	//parsing cache size
+	size, err := strconv.Atoi(args[1])
+	if err != nil || size <= 0 {
+		return nil, plugin.Error(pluginName, fmt.Errorf("invalid cache size: %s", args[1]))
+	}
+
+	// parsing check interval
+	interval, err := strconv.Atoi(args[2])
```

Comment:
Maybe we can treat this param as string and use `time.ParseDuration(args[2])` function?